internal/controllers/clusterclass/clusterclass_controller.go

```diff
@@ -173,6 +173,21 @@ func (r *Reconciler) reconcile(ctx context.Context, clusterClass *clusterv1.Clus
 		return ctrl.Result{}, kerrors.NewAggregate(errs)
 	}
 
+	// Ensure the variables are added to the ClusterClass status.
```

Comment:
WDYT about moving this above all this ref stuff? (Then the variables would be updated even if there's something wrong with the refs)

Or doesn't really matter as we don't bump the observedGeneration if we got an error anyway?

Reply:
Yeah this shouldn't matter. I'm fine either way as these are essentially parallel tasks, and I don't see a logical ordering really.

Reply:
As long as we don't use the .status.variables until observedGeneration == metadata.generation it doesn't make a difference agree.

I wonder a bit if it makes a difference for the user. But I guess it doesn't. They will either see the old or the new values, but they won't be used anyway.